File: WebQuanLyGiaiDau_NhomTD/Models/RewardProduct.cs
using System.ComponentModel.DataAnnotations;

namespace WebQuanLyGiaiDau_NhomTD.Models
{
    public class RewardProduct
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        [Range(0, int.MaxValue)]
        public int PointsCost { get; set; }

        public string? Description { get; set; }
    }
}

File: tournament_app/lib/models/reward_product.dart
class RewardProduct {
  final int id;
  final String name;
  final int pointsCost;
  final String? description;
  final String? imageUrl;

  RewardProduct({
    required this.id,
    required this.name,
    required this.pointsCost,
    this.description,
    this.imageUrl,
  });

  factory RewardProduct.fromJson(Map<String, dynamic> json) {
    return RewardProduct(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      pointsCost: json['pointsCost'] ?? 0,
      description: json['description'],
      imageUrl: json['imageUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'pointsCost': pointsCost,
      'description': description,
      'imageUrl': imageUrl,
    };
  }
}
